### ISSUE TITLE
[NBI] Tree crown detection and multi-class classification (DeepForest)

### ISSUE THREAD

**Issue by camappel** [alan-turing-institute/environmental-ds-book#251]
## What is the notebook about?

The notebook will demonstrate how to use the latest version of DeepForest's [CropModel](https://deepforest.readthedocs.io/en/latest/CropModels.html) for tree crown detection and multi-class classification using pre-trained models.

It will explore the performance of DeepForest's pre-trained models on different publicly available pre-processed data: 
- Homogeneous vs heterogeneous landscape
- Small distinct tree-crowns vs large dense canopies
- High vs low resolution images
- Tiles vs entire images

Additionally, it will cover post-processing steps to evaluate and visualise the model's predictions.

## Data Science Component

- [x] Exploration
- [ ] Preprocessing
- [ ] Modelling
- [x] Post-processing
- [ ] Other (e.g. Reproducibility): 

## Submission type

- [x] Standard
- [ ] Special Issue
- [ ] Other (e.g. CI2023 Reproducibility Challenge):

## Programming language

- [x] Python
- [ ] R
- [ ] Julia
- [ ] Other:

## Checklist:

- [x] Input data, pipeline and/or model are public with license/citation
- [x] The proposed notebook reuses existing codebase
- [x] The proposed notebook uses open-source packages
- [x] The proposed notebook is associated to existing publication(s)

## Additional information 
An EDS [notebook](https://edsbook.org/notebooks/gallery/15d986da-2d7c-44fb-af71-700494485def/notebook) already exists on tree crown detection using DeepForest. This notebook is different because it focusses on the latest version of DeepForest (1.4.0), which utilises model-chaining to classify detected tree crowns into multiple classes.

### Extension
DeepForest also allows uses to [train](https://deepforest.readthedocs.io/en/latest/training.html) the models: "The prebuilt models will always be improved by adding data from the target area. In our work, we have found that even an hour’s worth of carefully chosen hand-annotation can yield enormous improvements in accuracy and precision. 5-10 epochs of fine-tuning with the prebuilt model are often adequate." [Furthermore](https://github.com/weecology/DeepForest/discussions/677 
), "the backbone model is really meant for fine-tuning based on local annotations."

As such, there is the potential to extend the notebook to include fine-tuning the model by acquiring and pre-processing aerial RGB images using drones.

**Comment by acocac** [comment]
@camappel thanks for submitting the notebook idea. The purpose sounds great 🌳 🚀 

I have some comments about the structure and data/code to be showcased in the notebook. Please find them below.

I suggest focusing the notebook on modelling using pre-trained models. Where possible you should avoid training a model from scratch or fine-tuning it in the notebook because it could make unfeasible its reusability in Binder or practitioners without GPU resources.

I suggest showing the effect of different preprocessed data (homogeneous vs heterogeneous landscape, small vs large canopies) on the performance of such models. You can also consider using raw drone data and demonstrate some preprocessing, but keep in mind the image should be the smallest as possible to fit in the memory of Binder.

Can you complete the checklist? Previous contributors completed it before starting the next steps of the notebook.

Other comments:
> Tree crown detection and multi-class species classification using latest version of DeepForest's CropModel (https://deepforest.readthedocs.io/en/latest/CropModel.html#).
The above link doesn't work. Can you edit your comment to replace it with the correct URL?

**Comment by camappel** [comment]
Thanks for the feedback! I've updated the proposal to focus more on the evaluation of model performance on different types of pre-processed data, as opposed to pre-processing data and fine-tuning the model, given the emphasis on reproducibility (which would require a drone and GPU resources).

I have also left the original idea in Additional information as a potential extension to the project.

**Comment by acocac** [comment]
@camappel cool - thanks for implementing the suggestions and complete the checklist. Everything looks good to me!

I'm happy to support the submission of the notebook. This means you should move to [the preparation stage](https://edsbook.org/publishing/guidelines/guidelines-authors#notebook-repository). 

Please let me know if you find issues in the available templates or missing pieces in the guidelines.

**Comment by camappel** [comment]
Brilliant, thanks. I've forked the repo and opened a PR [here](https://github.com/camappel/DeepForest/pull/1). The Binder environment deployed correctly after a couple attempts, presumably because of the Docker image size. 

Thanks for the clear templates/guidelines; I'll tag you when the minimal working version of the notebook is ready.

**Comment by acocac** [comment]
> Brilliant, thanks. I've forked the repo and opened a PR [here](https://github.com/camappel/DeepForest/pull/1). The Binder environment deployed correctly after a couple attempts, presumably because of the Docker image size.
> 
> Thanks for the clear templates/guidelines; I'll tag you when the minimal working version of the notebook is ready.

Awesome - Binder is quite unstable because it has been on reduced capacity, but it's a useful resource. Anyone can check the status [here](https://mybinder.readthedocs.io/en/latest/about/status.html#mybinder-org-status). 

Glad to hear the guidelines are clear. Looking forward to commenting on the first working version!

**Comment by camappel** [comment]
Unfortunately, I do not think this notebook is feasible in it's current form. The binder environment is running properly in [this branch](https://github.com/camappel/DeepForest/tree/cam) of my forked repo, and I'm at the stage of identifying different classes of the detected tree crowns. 

However, the [updated docs](https://deepforest.readthedocs.io/en/latest/prebuilt.html#crop-classifiers) and various [issues](https://github.com/weecology/DeepForest/pull/675) make it clear that the prebuilt crop classifier has only been trained to classify alive vs dead trees. From my understanding, the package is designed to allow the fine-tuning of the prebuilt models based on new annotations; but even so, the model would [probably not](https://github.com/weecology/DeepForest/issues/662) be able to distinguish different species of trees from aerial photos. It could be feasible given sufficient labelled data and compute resources for training, but I believe this would be outside of the EDS' scope.

Therefore, it is not suitable to the task proposed in the notebook: to classify different species of trees in different contexts from aerial RGB photos using the pre-trained crop classifier model.

As such, I would like to close this notebook issue and propose a new one, also focussing on deepforest. In particular, I could focus on one of the other prebuilt models with more general applications, such as single-class bird or [live-stock](https://github.com/weecology/DeepForest/pull/756) detection.

Please let me know your thoughts!
